### PR TITLE
tests(runner): drop usages of legacy runner

### PR DESCRIPTION
**Pull request by adamraine**
- Stop using `Runner._gatherArtifactsFromBrowser` in `gatherFn`
- Convert `LegacyResolvedConfig.fromJson` calls to new `initializeConfig`
- Remove some obsolete tests (I will call these out individually)

### CODE REVIEW

**Comment by connorjclark** [comment]
Previously the new runner was being given a gatherFn that just ran the legacy runner to do the collection (which we mocked the hell out of to get what we want). This PR stops using legacy runner to fill out the gatherFn and instead uses various bits of the navigation runner implementation. Also, legacy config stuff is replaced with calls to `initializeConfig`.

Is that everything?

File: core/test/runner-test.js [hunk 1]
```diff
@@ -283,25 +327,9 @@ describe('Runner', () => {
 
     return runGatherAndAudit(createGatherFn(url),
         {resolvedConfig, driverMock, computedCache: new Map()}).then(_ => {
-      expect(gatherRunnerRunSpy).toHaveBeenCalled();
-      assert.ok(typeof resolvedConfig.passes[0].gatherers[0] === 'object');
-    });
-  });
-
-  it('rejects when given neither passes nor artifacts', async () => {
```
Comment (adamraine) [inline core/test/runner-test.js:291]: The FR runner methods are not responsible for checking this

File: core/test/runner-test.js [hunk 2]
```diff
@@ -343,53 +371,33 @@ describe('Runner', () => {
...
     return runGatherAndAudit({}, {resolvedConfig, computedCache: new Map()}).then(results => {
+      expect(mockGatherImpl).not.toHaveBeenCalled();
       const audits = results.lhr.audits;
       assert.equal(audits['user-timings'].displayValue, '2 user timings');
       assert.deepStrictEqual(audits['user-timings'].details.items.map(i => i.startTime),
         [0.002, 1000.954]);
     });
   });
 
-  it('rejects when given an invalid trace artifact', async () => {
```
Comment (adamraine) [inline core/test/runner-test.js:363]: Trace artifact is not given special treatment by the FR runner anymore

File: core/test/runner-test.js [hunk 3]
```diff
@@ -838,55 +890,19 @@ describe('Runner', () => {
       expect(lhr.runtimeError.code).toEqual(NO_FCP.code);
       expect(lhr.runtimeError.message).toMatch(/did not paint any content.*\(NO_FCP\)/);
     });
-
-    it('includes a pageLoadError runtimeError over any gatherer runtimeErrors', async () => {
```
Comment (adamraine) [inline core/test/runner-test.js:842]: The general purpose runner is not responsible for this logic

**Comment by adamraine** [comment]
> Is that everything?

Yes + removing some obsolete tests

File: core/test/runner-test.js (hunk 3 above)
Comment (adamraine) [reply core/test/runner-test.js:842]: `navigation-runner.js`

File: core/test/runner-test.js (hunk 1 above)
Comment (adamraine) [reply core/test/runner-test.js:291]: `navigation-runner.js` (sort of)

It rejects when there are no "config navigations" but there should always be 1 even if the `artifacts` array on the config doesn't exist or is empty.

**Comment by Tonyadelgado1982** [comment]
tonya Delgado

**Pull request merged**